src/commonMain/kotlin/fr/acinq/lightning/channel/states/WaitForAcceptChannel.kt

```diff
@@ -52,40 +51,64 @@ data class WaitForAcceptChannel(
                             val remoteFundingPubkey = accept.fundingPubkey
                             val dustLimit = accept.dustLimit.max(init.localParams.dustLimit)
                             val fundingParams = InteractiveTxParams(channelId, true, init.fundingAmount, accept.fundingAmount, remoteFundingPubkey, lastSent.lockTime, dustLimit, lastSent.fundingFeerate)
-                            when (val fundingContributions = FundingContributions.create(channelKeys, keyManager.swapInOnChainWallet, fundingParams, init.walletInputs)) {
+                            when (val lease = LiquidityAds.validateLease(
+                                lastSent.requestFunds,
+                                staticParams.remoteNodeId,
+                                channelId,
+                                fundingParams.fundingPubkeyScript(channelKeys),
+                                accept.fundingAmount,
+                                lastSent.fundingFeerate,
+                                accept.willFund
+                            )) {
```

Comment:
Just for my understanding: adding this code here (and commit b5c187db6fc4079984838fc4b1eba9a2ef421a87 in general) does not change anything in terms of trust for the fee computation and overall funding amount from the point of view of the buyer right? Meaning that those were already computed in `fundingParams` and enforced by `InteractiveTx`?

Reply:
Exactly, this is used only when the wallet initiates funding because they have swap-in-potentiam inputs, so the fees are paid atomically during `interactive-tx` and don't require trust on any side.